internal/db/disk/submission.go

```diff
@@ -316,7 +336,8 @@ func (this *backend) getUserSubmissionDir(courseID string, assignmentID string,
 	return filepath.Join(this.getCourseDirFromID(courseID), model.SUBMISSIONS_DIRNAME, assignmentID, user)
 }
 
-// Get the short id of the most recent submission (or empty string if there are no submissions).
+// Get the short ID of the most recent submission (or empty string if there are no submissions).
+// The most recent submission has the largest ID for the user.
```

Comment:
The semantics for IDs needs to be laid out at the database level (right now you are at the database backend level).
For example, if we have another database backend it needs to follow the same rules.

Note that a well-placed comment can be sufficient for this
(if it is not something that can be easily enforced with code).